File: Tree-shaped Deep Forest/Binary Tree/Q1. Root Equals Sum of Children/Q1_RootEqualsSumofChildren.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;          // value stored in the node
 *     TreeNode left;    // reference to left child
 *     TreeNode right;   // reference to right child
 *     TreeNode() {}     
 *     TreeNode(int val) { this.val = val; }  // constructor to set node value
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;      // assign value
 *         this.left = left;    // assign left child
 *         this.right = right;  // assign right child
 *     }
 * }
 */

class Solution {
    public boolean checkTree(TreeNode root) {

        // check if root value equals sum of left and right child values
        return root.val == root.left.val + root.right.val;
    }
}

File: Tree-shaped Deep Forest/Binary Tree/Q1. Root Equals Sum of Children/Q1_RootEqualsSumofChildren.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val      # value of the node
#         self.left = left    # reference to left child
#         self.right = right  # reference to right child

class Solution(object):
    def checkTree(self, root):
        """
        :type root: Optional[TreeNode]   # root of the binary tree
        :rtype: bool                     # return True or False
        """

        # check if root value equals sum of its left and right child values
        return root.val == root.left.val + root.right.val